### ISSUE TITLE
awful.tag.viewmore should use screen from tags

### ISSUE THREAD

**Issue by blueyed** [awesomeWM/awesome#1883]
Since `tags` is meant to be a table with tag objects (is this correct?!), the screen should be taken from there.
This should happen at least by default, but it seems like there is no use for specifying another screen?!

Code for reference:

```lua
--- View only a set of tags.
-- @function awful.tag.viewmore
-- @param tags A table with tags to view only.
-- @param[opt] screen The screen of the tags.
function tag.viewmore(tags, screen)
    screen = get_screen(screen or ascreen.focused())
    local screen_tags = screen.tags
    for _, _tag in ipairs(screen_tags) do
        if not gtable.hasitem(tags, _tag) then
            _tag.selected = false
        end
    end
    for _, _tag in ipairs(tags) do
        _tag.selected = true
    end
    screen:emit_signal("tag::history::update")
end
```

Source: https://github.com/awesomeWM/awesome/blob/cbb8fd7ac408d229233a3b5490adf35d58a85d79/lib/awful/tag.lua#L1275-L1291

**Comment by blueyed** [comment]
@Elv13 
Do you agree?

**Comment by Elv13** [comment]
Right, the fallback screen should be queried from the tags, but `screen` should still be honored when set. There is nothing in the C core that really enforce that the tags are from the same screen (on purpose, since v3.4). This is why the code for the function is as ugly as it is. But I agree the default here is wrong.

I also don't think someone will notice, so that doesn't count as a API break to me.

**Comment by blueyed** [comment]
Cool.
But how is `screen` useful in case it is different from the tag(s) screen?
Should it get ignored then?

I came up with https://github.com/awesomeWM/awesome/pull/1884 (before reading your comment).

**Comment by blueyed** [comment]
I've noticed this through https://github.com/awesomeWM/awesome/issues/1886.
